Microsoft.Toolkit.Uwp.UI.Controls/BladeView/BladeItemAutomationPeer.cs

```diff
@@ -59,25 +61,97 @@ protected override string GetClassNameCore()
         /// </returns>
         protected override string GetNameCore()
         {
-            int? index = this.OwnerBladeItem.ParentBladeView.GetBladeItems().ToList().IndexOf(this.OwnerBladeItem);
+            string name = AutomationProperties.GetName(this.OwnerBladeItem);
+            if (!string.IsNullOrEmpty(name))
+            {
+                return name;
+            }
 
-            string name = base.GetNameCore();
+            name = this.OwnerBladeItem.Name;
             if (!string.IsNullOrEmpty(name))
             {
-                return $"{name} {index}";
+                return name;
             }
 
-            if (this.OwnerBladeItem != null && !string.IsNullOrEmpty(this.OwnerBladeItem.Name))
+            TextBlock textBlock = this.OwnerBladeItem.FindDescendant<TextBlock>();
```

Comment:
The `BladeItem.Header` should be the 'Title' of the Blade, do we want to try and explicitly convert that to a string as a first step?

Reply:
Sorted! Apologies for the delay on sorting this out.